### ISSUE TITLE
Move logging configuration under `logging` key

### ISSUE THREAD

**Issue by aanand** [docker/compose#2503]
Right now we have `log_driver` and `log_opt`. It'd be cleaner to have them both under a `logging` key:

```yaml
logging:
  driver: syslog
  options:
    syslog-address: "tcp://192.168.0.42:123"
```

An additional advantage: overriding a service's `logging` option would be cleaner, as you only have to override a single key (you're never going to want to override `log_driver` but *not* `log_opt`, because options are driver-specific).

**Comment by dbonev** [comment]
#dibs

**Comment by aanand** [comment]
Closed by #2626
